### PR TITLE
REF: share BooleanArray tests with MaskedArray

### PR DESCRIPTION
**Pull request by jbrockmendel**
mirror the layout in the pyarrow tests.

### CODE REVIEW

File: pandas/tests/groupby/test_function.py [hunk 1]
```diff
@@ -1358,6 +1358,30 @@ def test_apply_to_nullable_integer_returns_float(values, function):
...
 
+@pytest.mark.parametrize("min_count", [0, 10])
+def test_groupby_sum_mincount_boolean(min_count):
+    b = True
+    a = False
+    c = b
+    na = np.nan
+    dfg = pd.array([b, b, na, na, a, a, b, c], dtype="boolean")
+
+    df = DataFrame({"A": [1, 1, 2, 2, 3, 3, 1], "B": dfg[:-1]})
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_function.py:1369]: nit: I see this was copied from a more generic test, should we clean up here? I.e. remove occurrences of `c` and `dfg[:-1] -> dfg`

**Review by rhshadrach — COMMENTED** [review]
lgtm, just a question

File: pandas/tests/groupby/test_function.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/groupby/test_function.py:1369]: good catch, updated

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**